apps/src/code-studio/assets/SpriteLabAnimationUpload.jsx

```diff
@@ -0,0 +1,590 @@
+import React from 'react';
+import color from '@cdo/apps/util/color';
+import {makeEnum} from '@cdo/apps/utils';
+import ValidationStep, {Status} from '../../lib/ui/ValidationStep';
+import {
+  getManifest,
+  getLevelAnimationsFiles,
+  uploadAnimationToAnimationLibrary,
+  uploadMetadataToAnimationLibrary
+} from '@cdo/apps/assetManagement/animationLibraryApi';
+
+const EXTENSION_CHECK = 'extensionError';
+const FILENAME_CHECK = 'filenameError';
+const FILESIZE_CHECK = 'filesizeError';
+const ANIMATION_EXISTS_CHECK = 'animationExistsError';
+const AnimationLocations = makeEnum('library', 'level');
+
+const initialSpriteLabLibrariesState = {
+  spriteLabLevelAnimations: [],
+  spriteLabLibraryCategories: [],
+  spriteLabLibraryAnimations: []
+};
+
+const newImageState = {
+  fileData: null,
+  filename: '',
+  filePreviewURL: ''
+};
+
+const imageChecksInitialState = {
+  [EXTENSION_CHECK]: Status.WAITING,
+  [FILENAME_CHECK]: Status.WAITING,
+  [FILESIZE_CHECK]: Status.WAITING,
+  [ANIMATION_EXISTS_CHECK]: Status.WAITING
+};
+
+const initialUserInputMetaData = {
+  availability: '',
+  category: '',
+  aliases: []
+};
+
+const initialGeneratedMetadata = {
+  frameSize: {x: NaN, y: NaN},
+  destination: '',
+  metadata: ''
+};
+
+const initialComponentState = {
+  ...newImageState,
+  ...imageChecksInitialState,
+  ...initialGeneratedMetadata
+};
+
+const maxSize = 800;
+const minSize = 100;
+
+// Levelbuilder tool for adding animations to the Spritelab animation library.
+export default class AnimationUpload extends React.Component {
```

Comment:
I like renaming this to align with our sprite -> animation naming decisions, but I'm hesitant about renaming the file to SpriteLabAnimationUpload because using SpriteLab specific variable naming. Most (if not all) of this component can be reused for GameLab animation management in the future. I'd love to leave that path open for an extension of this work. Happy to chat a bit about feasibility and renaming!

Reply:
How about just AnimationUpload.jsx?

Reply:
:+1:

Reply:
Reverted to SpriteUpload.jsx for this PR. As discussed, we'll handle any renaming of files when we do the follow-up work to break this into separate components.